lib/services/wallet_connect_service.dart

```diff
@@ -355,6 +361,46 @@ class WalletConnectService {
           }
         });
 
+    web3Wallet!.registerRequestHandler(
+        chainId: Config.walletConnectChainId,
+        method: WcMethods.wSendTransaction,
+        handler: (topic, args) async {
+          final sessionRequest = web3Wallet!.pendingRequests
+              .getAll()
+              .where((e) =>
+                  e.method == WcMethods.wSendTransaction && e.topic == topic)
+              .last;
+
+          late RequestSendTransactionEvent event;
+
+          if (sendTransactionHandler == null) {
+            throw "sendQubicHandler is not set";
+          }
+          try {
+            event = RequestSendTransactionEvent.fromMap(
+                args, topic, sessionRequest.id);
+            event.validateOrThrow();
+
+            if (web3Wallet!.getActiveSessions().containsKey(topic)) {
+              event.setPairingMetadata(
+                  web3Wallet!.getActiveSessions()[topic]!.peer.metadata);
+            } else {
+              throw "Session not found";
+            }
+
+            return web3Wallet!.respondSessionRequest(
+                topic: topic,
+                response: JsonRpcResponse(
+                    id: sessionRequest.id,
+                    result: await sendTransactionHandler!(event)));
+          } catch (e) {
+            if (e is JsonRpcError) {
+              rethrow;
+            }
+            throw JsonRpcError(code: -2, message: e.toString());
+          }
```

Comment:
@ahmed-tarek-salem  what his -2 is? if we have a list of error codes, we should define as constant and document for the DApps